crates/forge/src/runner.rs

```diff
@@ -407,129 +409,134 @@ impl<'a> ContractRunner<'a> {
 
     #[instrument(name = "invariant-test", skip_all)]
     pub fn run_invariant_test(
-        &mut self,
+        &self,
         runner: TestRunner,
         setup: TestSetup,
         invariant_config: InvariantConfig,
-        functions: Vec<&Function>,
+        func: &Function,
         known_contracts: Option<&ContractsByArtifact>,
-        identified_contracts: ContractsByAddress,
-    ) -> Vec<TestResult> {
-        trace!(target: "forge::test::fuzz", "executing invariant test with invariant functions {:?}",  functions.iter().map(|f|&f.name).collect::<Vec<_>>());
+        identified_contracts: &ContractsByAddress,
+    ) -> TestResult {
+        trace!(target: "forge::test::fuzz", "executing invariant test for {:?}", func.name);
         let empty = ContractsByArtifact::default();
         let project_contracts = known_contracts.unwrap_or(&empty);
         let TestSetup { address, logs, traces, labeled_addresses, .. } = setup;
 
         // First, run the test normally to see if it needs to be skipped.
-        if let Err(EvmError::SkipError) = self.executor.execute_test::<(), _, _>(
+        if let Err(EvmError::SkipError) = self.executor.clone().execute_test::<(), _, _>(
             self.sender,
             address,
-            functions[0].clone(),
+            func.clone(),
             (),
             0.into(),
             self.errors,
         ) {
-            return vec![TestResult {
+            return TestResult {
                 status: TestStatus::Skipped,
                 reason: None,
                 decoded_logs: decode_console_logs(&logs),
                 traces,
                 labeled_addresses,
-                kind: TestKind::Standard(0),
+                kind: TestKind::Invariant { runs: 1, calls: 1, reverts: 1 },
                 ..Default::default()
-            }]
+            }
         };
 
         let mut evm = InvariantExecutor::new(
-            &mut self.executor,
+            self.executor.clone(),
             runner,
             invariant_config,
             &identified_contracts,
             project_contracts,
         );
 
         let invariant_contract =
-            InvariantContract { address, invariant_functions: functions, abi: self.contract };
+            InvariantContract { address, invariant_functions: func, abi: self.contract };
 
-        let Ok(InvariantFuzzTestResult { invariants, cases, reverts, last_run_inputs }) =
-            evm.invariant_fuzz(&invariant_contract)
-        else {
-            return vec![]
```

Comment:
This error was just completely ignored for some reason. This gets thrown by `prepare_fuzzing`

Reply:
Yeah, I think there's a few more cases of this happening. I have a pending refactor of another error being ignored on invariants